### PR TITLE
OOP School Library: Class Decoration

### PR DESCRIPTION
**Pull request by Shisui6**
## Summary 🧾
- Create a decorator class to decorate the person class

### CODE REVIEW

File: lib/decorate.rb [hunk 1]
```diff
@@ -0,0 +1,28 @@
+class Nameable
+  def correct_name
+    raise NotImplementedError, "#{self.class} has not implemented '#{__method__}'"
+  end
+end
+
+class Decorator < Nameable
+  def initialize(nameable)
```
Comment (MahmoudBakr23) [inline lib/decorate.rb:10]: - [ ] Please define the relationship between the Decorator and Nameable, it has to be done by using the accessor's attributes such as `attr_accessor :nameable`. 🙏

File: lib/decorate.rb [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    super
```
Comment (MahmoudBakr23) [inline lib/decorate.rb]: - [ ] Please complete the `super` method by modifying it to be `super()`

File: lib/person.rb [hunk 3]
```diff
@@ -1,10 +1,12 @@
 require 'securerandom'
+require './decorate'
 
-class Person
+class Person < Nameable
   attr_accessor(:name, :age)
   attr_reader :id
 
   def initialize(age, name = 'Unknown', parent_permission: true)
+    super
```
Comment (MahmoudBakr23) [inline lib/person.rb]: - [ ] Please complete the `super` method by modifying it to be `super()`

**Review by MahmoudBakr23 — CHANGES_REQUESTED** [review]
Hello @Shisui6 😃, 

Great work and effort you put into this project so far.

## Highlights 🚀 

There are some excellent points that you have made perfectly. Keep up! 💪
- Well done on creating the correct decorating classes for this milestone so far
- Well done with the class (constructor) and custom methods
- Descriptive PR and README file
- Correct Linters

However, there is always room for improvement.

> Keep in mind that your long hours of work are appreciated. However, the code review process is for improving the code you wrote, and not for criticizing it. 😄

 ## Required Changes ♻️
_Please check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

- Great job so far separating your classes. However, it's cleaner and more optimized if you structured your files and separated your code into different files, one for each class. 🙏 

Cheers and Happy coding!👏👨‍💻💻

Feel free to mention @MahmoudBakr23 to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Gambit142 — APPROVED** [review]
Hi @Shisui6 👋,

## Status: Approved 🟢   

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉

### Highlights
- All linters checks passed ✔️ 
- Created Nameable class ✔️ 
- Created base decorator ✔️ 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**